Content.Server/Shipyard/Systems/ShipyardSystem.Consoles.cs

```diff
@@ -333,12 +319,9 @@ public void OnSellMessage(EntityUid uid, ShipyardConsoleComponent component, Shi
 
         RemComp<ShuttleDeedComponent>(targetId);
 
-        if (ShipyardConsoleUiKey.Security == (ShipyardConsoleUiKey) args.UiKey)
-            channel = component.SecurityShipyardChannel;
-
-        if (ShipyardConsoleUiKey.BlackMarket == (ShipyardConsoleUiKey) args.UiKey || ShipyardConsoleUiKey.Syndicate == (ShipyardConsoleUiKey) args.UiKey) // Unhardcode this please
+        if (component.SalesTax != 0f)
```

Comment:
```suggestion
        if (component.SalesTax > 0.0f)
```
Equality check doesn't play nicely with NaN.  If we need bonuses on selling ships, this needs revision.

Reply:
Actually, since this same check gets repeated a few times, it'd be nice to put into a function, `HasValidSalesTax()` or the like.

Reply:
I moved sales tax into a new method called `CalculateSalesTax`, which uses `float.IsFinite` to exclude NaN and infinities.